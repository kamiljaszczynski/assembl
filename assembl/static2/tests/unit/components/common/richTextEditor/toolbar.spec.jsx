import React from 'react';
import renderer from 'react-test-renderer';
import { EditorState } from 'draft-js';

import Toolbar from '../../../../../js/app/components/common/richTextEditor/toolbar';

describe('Toolbar component', () => {
  it('should render a toolbar', () => {
    const buttonsConfig = [
      {
        id: 'bold',
        icon: 'text-bold',
        label: 'Bold',
        type: 'style',
        style: 'BOLD'
      }
    ];
    const editorState = EditorState.createEmpty();
    const focusEditorSpy = jest.fn(() => {});
    const onChangeSpy = jest.fn(() => {});
    const component = renderer.create(
      <Toolbar buttonsConfig={buttonsConfig} editorState={editorState} focusEditor={focusEditorSpy} onChange={onChangeSpy} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});