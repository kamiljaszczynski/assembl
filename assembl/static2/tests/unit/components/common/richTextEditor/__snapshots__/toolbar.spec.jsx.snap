// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component should render a toolbar 1`] = `
<div
  className="editor-toolbar"
>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    title="Bold"
    type="button"
  >
    <span
      className="assembl-icon-text-bold"
    />
  </button>
</div>
`;
